src/libsyntax/visit.rs

```diff
@@ -162,6 +162,9 @@ pub trait Visitor<'ast>: Sized {
     fn visit_fn_header(&mut self, _header: &'ast FnHeader) {
         // Nothing to do
     }
+    fn visit_fn_decl(&mut self, _decl: &'ast FnDecl) {
+        // Nothing to do
+    }
```

Comment:
Could you avoid this change?
`Visitor` methods are supposed to be customization points that are called from the generic visiting infrastructure, which is not the case here.
It can be just a `fn check_fn_decl` method internal to AST validation.